src/model/SVGFile.java

```diff
@@ -21,21 +28,25 @@
 	private static String filepath = "files"+File.separator+"%s.svg";
 
 	
-	public SVGFile(String filnename, String xmlcontent) {
-		this.filnename = filnename;
+	public SVGFile(String fileName, String xmlcontent, Canvas canvas) {
```

Comment:
not sure about that.
Actually SVGFile should be FileCreation (not only SVG) and here the canvas deliver some xml, so it would be "written" inside the "xmlcontent" which should just be content actually